app/models/lti_configuration.rb

```diff
@@ -32,12 +34,38 @@ def context_membership_url(use_cache: true, nonce: nil)
     message = message_store.get_message(nonce)
     return nil unless message
 
-    membership_url = message.custom_params["custom_context_memberships_url"]
+    membership_url = message.custom_params[lms_settings.context_memberships_url_key]
     return nil unless membership_url
 
     self[:context_membership_url] = membership_url
     save!
 
     membership_url
   end
+
+  def xml_configuration(launch_url)
+    return nil unless supports_autoconfiguration
+
+    builder = GitHubClassroom::LTI::ConfigurationBuilder.new("GitHub Classroom", launch_url)
+
+    builder.add_attributes(
+      description: "Sync your GitHub Classroom organization with your Learning Management System.",
+      icon: "https://classroom.github.com/favicon.ico",
+      vendor_name: "GitHub Classroom",
+      vendor_url: "https://classroom.github.com/"
+    )
+
+    builder.add_vendor_attributes(lms_settings.vendor_domain, lms_settings.vendor_attributes)
+    builder.to_xml
+  end
+
+  private
+
+  def lms_settings
```

Comment:
This is where the only interesting thing in this PR happens, which is dynamically determining what kind of `LtiConfiguration::Settings` to use based on the already existing `lms_type`.